### PR TITLE
Fix bugs in distributions

### PR DESCRIPTION
**Pull request by lostella**
*Issue #, if available:* #91 

*Description of changes:*
There was some shape problem with the NegativeBinomialOutput class. I added a test for all Output classes, and fixed a couple of related issues.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-92/1 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-92/1/index.html

File: test/distribution/test_distribution_output_shapes.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    data: Tensor,
+    scale: List[Union[None, Tensor]],
+    expected_batch_shape: Tuple,
+    expected_event_shape: Tuple,
+):
+    args_proj = distr_out.get_args_proj()
+    args_proj.initialize()
+
+    args = args_proj(data)
+
```
Comment (lostella) [inline test/distribution/test_distribution_output_shapes.py]:
```suggestion

assert distr_out.event_shape == expected_event_shape

```

File: test/distribution/test_distribution_output_shapes.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+assert distr_out.event_shape == expected_event_shape
```
Comment (lostella) [inline test/distribution/test_distribution_output_shapes.py]:
```suggestion
    assert distr_out.event_shape == expected_event_shape
```

**Comment by szha** [comment]
Job PR-92/2 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-92/2/index.html

**Comment by szha** [comment]
Job PR-92/3 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-92/3/index.html

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/gluon-ts/pull/92?src=pr&el=h1) Report
> Merging [#92](https://codecov.io/gh/awslabs/gluon-ts/pull/92?src=pr&el=desc) into [master](https://codecov.io/gh/awslabs/gluon-ts/commit/b009be4b13d0884790889a6da2cc731ef0dd9098?src=pr&el=desc) will **increase** coverage by `0.4%`.
> The diff coverage is `100%`.


[Impacted file tree graph](https://codecov.io/gh/awslabs/gluon-ts/pull/92?src=pr&el=tree)

```diff
@@            Coverage Diff            @@
##           master      #92     +/-   ##
=========================================
+ Coverage   78.63%   79.03%   +0.4%     
=========================================
  Files         110      110             
  Lines        6285     6288      +3     
=========================================
+ Hits         4942     4970     +28     
+ Misses       1343     1318     -25
```

| [Impacted Files](https://codecov.io/gh/awslabs/gluon-ts/pull/92?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/gluonts/distribution/uniform.py](https://codecov.io/gh/awslabs/gluon-ts/pull/92/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL3VuaWZvcm0ucHk=) | `90.9% <100%> (+6.81%)` | :arrow_up: |
| [src/gluonts/distribution/neg\_binomial.py](https://codecov.io/gh/awslabs/gluon-ts/pull/92/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL25lZ19iaW5vbWlhbC5weQ==) | `98.33% <100%> (+8.33%)` | :arrow_up: |
| [src/gluonts/distribution/piecewise\_linear.py](https://codecov.io/gh/awslabs/gluon-ts/pull/92/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL3BpZWNld2lzZV9saW5lYXIucHk=) | `100% <100%> (+12.5%)` | :arrow_up: |
| [src/gluonts/distribution/mixture.py](https://codecov.io/gh/awslabs/gluon-ts/pull/92/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL21peHR1cmUucHk=) | `97.36% <0%> (+1.31%)` | :arrow_up: |
| [src/gluonts/distribution/multivariate\_gaussian.py](https://codecov.io/gh/awslabs/gluon-ts/pull/92/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL211bHRpdmFyaWF0ZV9nYXVzc2lhbi5weQ==) | `98.43% <0%> (+1.56%)` | :arrow_up: |
| [src/gluonts/distribution/gaussian.py](https://codecov.io/gh/awslabs/gluon-ts/pull/92/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL2dhdXNzaWFuLnB5) | `91.3% <0%> (+2.17%)` | :arrow_up: |
| [src/gluonts/distribution/laplace.py](https://codecov.io/gh/awslabs/gluon-ts/pull/92/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL2xhcGxhY2UucHk=) | `100% <0%> (+4.65%)` | :arrow_up: |

**Pull request merged**